rosidl_generator_cpp/test/test_traits.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "rosidl_generator_cpp/msg/empty.hpp"
+#include "rosidl_generator_cpp/msg/string.hpp"
+#include "rosidl_generator_cpp/srv/complex_type_zero_fill.hpp"
+
+using rosidl_generator_traits::is_message;
+using rosidl_generator_traits::is_service;
+using rosidl_generator_traits::is_service_request;
+using rosidl_generator_traits::is_service_response;
+
+// Empty testing struct
+struct Message {};
+
+// Empty testing struct, with template instantiation
+struct Message2 {};
+
+template<>
+struct rosidl_generator_traits::is_message<Message2>: std::true_type {};
+
+TEST(Test_rosidl_generator_traits, is_message) {
+  // A message is not a service
+  ASSERT_TRUE(is_message<rosidl_generator_cpp::msg::Empty>());
```

Comment:
nit: using `EXPECT_*` instead, we'd see multiple failures instead of just the first to fail.